[Docs]  Add setup content from GET (#1770)
grafana/tempo: docs/tempo/website/getting-started/_index.md

---
title: Get started
menuTitle: Get started with Grafana Tempo
weight: 150
---

# Get started with Grafana Tempo

Distributed tracing visualizes the lifecycle of a request as it passes through
a set of applications.
There are a few components that must be configured in order to get a
working distributed tracing visualization.

This document discusses the four major pieces necessary to build out a tracing system:

> Instrumentation, Pipeline, Backend and Visualization.

Reviewer: ```suggestion
Instrumentation, pipeline, backend, and visualization.
```

> If one were to build a diagram laying
> out these pieces, it may look something like this:

Reviewer: ```suggestion
The following diagram illustrates a tracing system configuration:
```

Author: Good catch. I read this and thought passive voice, should be changed. Thank you.

---

> title: Get started
> menuTitle: Get started with Grafana Tempo

Reviewer 2: ```suggestion
title: Get started with Grafana Tempo
menuTitle: Get started
```
See https://raw.githubusercontent.com/grafana/mimir/main/docs/sources/operators-guide/get-started/_index.md.

* [OpenTelemetry Java auto-instrumentation](https://github.com/open-telemetry/opentelemetry-java-instrumentation)
* [OpenTelemetry .NET auto-instrumentation](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation)
* [OpenTelemetry Python auto-instrumentation](https://github.com/open-telemetry/opentelemetry-python-contrib)

> **Note**: Check out the [instrumentation references]({{< relref "./instrumentation" >}}) to learn how to instrument your
> favorite language for distributed tracing.

## Pipeline (Grafana Agent)

Once your application is instrumented for tracing, the traces need to be sent
to a backend for storage and visualization. You can build a tracing pipeline that
offloads spans from your application, buffers them, and eventually forwards them to a backend.
Tracing pipelines are optional (most clients can send directly to Tempo), but the pipelines
become more critical the larger and more robust your tracing system is.

The Grafana Agent is a service that is deployed close to the application, either on the same node or
within the same cluster (in Kubernetes) to quickly offload traces from the application and forward them to
a storage backend.

> The Grafana Agent also abstracts features like trace batching and backend routing away from the client.

Reviewer 3: Does it also add retry logic?

Author: @hedss  do you know if it does?

Reviewer 4: Yes, this uses the full OTel retry on failure configuration options. So maybe we should say:
```
The Grafana Agent also abstracts features like trace batching to a remote trace backend store, including retries on write failures.
```

Author: Thank you. I've updated the text to reflect your suggestion.

> The Grafana Agent also abstracts features like trace batching and backend routing away from the client.

Author: ```suggestion
The Grafana Agent also abstracts features like trace batching to a remote trace backend store, including retries on write failures.
```

> ## 1. Instrumentation

Reviewer: Can we take the numbers out of the heading, and instead consolidate steps?

Author: Done.

> ## 1. Instrumentation

Author: ```suggestion
## Instrumentation
```

* [Zipkin](https://zipkin.io/pages/tracers_instrumentation)
* [OpenTelemetry](https://opentelemetry.io/docs/concepts/instrumenting/)

> ### OpenTelemetry Auto Instrumentation

Reviewer: ```suggestion
### OpenTelemetry automatic instrumentation
```
